extensions/sql-migration/src/dialog/validationResults/validateIrDialog.ts

```diff
@@ -0,0 +1,676 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import * as constants from '../../constants/strings';
+import { validateIrDatabaseMigrationSettings, validateIrSqlDatabaseMigrationSettings } from '../../api/azure';
+import { MigrationStateModel, MigrationTargetType, ValidateIrState, ValidationResult } from '../../models/stateMachine';
+import { EOL } from 'os';
+import { IconPathHelper } from '../../constants/iconPathHelper';
+
+const DialogName = 'ValidateIrDialog';
+
+enum ValidationResultIndex {
+	message = 0,
+	icon = 1,
+	status = 2,
+	errors = 3,
+	state = 4,
+}
+
+export const ValidationStatusLookup: constants.LookupTable<string | undefined> = {
+	[ValidateIrState.Canceled]: constants.VALIDATION_STATE_CANCELED,
+	[ValidateIrState.Failed]: constants.VALIDATION_STATE_FAILED,
+	[ValidateIrState.Pending]: constants.VALIDATION_STATE_PENDING,
+	[ValidateIrState.Running]: constants.VALIDATION_STATE_RUNNING,
+	[ValidateIrState.Succeeded]: constants.VALIDATION_STATE_SUCCEEDED,
+	default: undefined
+};
+
+export class ValidateIrDialog {
+	private _canceled: boolean = true;
+	private _dialog: azdata.window.Dialog | undefined;
+	private _disposables: vscode.Disposable[] = [];
+	private _isOpen: boolean = false;
+	private _model!: MigrationStateModel;
+	private _resultsTable!: azdata.TableComponent;
+	private _startLoader!: azdata.LoadingComponent;
+	private _startButton!: azdata.ButtonComponent;
```

Comment:
I'm having some issues with the "start validation " button, which I have filed a bug for: https://msdata.visualstudio.com/Database%20Systems/_workitems/edit/2126179

Reply:
I also have some general feedback about this button though. If I understand correctly it seems that the validation happens as soon as this validation dialog is opened, so is there really a need for a start button at all? Or should it maybe be a **re**start button instead? 

For example, as soon as I open the validation dialog this is the first thing I see: 
![image](https://user-images.githubusercontent.com/11844501/207174096-7f585de0-b2c5-4c5f-8c95-cae63fdf7019.png)


It's a little confusing. I can tell it's already been run (because everything succeeded) but it's confusing why there's a start button still showing. Does this mean restart? Or does it mean start for the first time? Might not be immediately obvious for a first time user